ZENKO-4363 add azure archive tests

welansari opened the pull request:
Issue: ZENKO-4363

Commit:
add azure archive e2e tests in ctst

Issue: ZENKO-4363

Commit:
pass azure archive tests specific params in world object

Issue: ZENKO-4363

Commit:
configure tar backend for tests

bert-e commented:
# Hello kerkesni,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/zenko/doc/user).





*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4363 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.7.1`


Please check the `Fix Version/s` of ZENKO-4363, or the target
branch of this pull request.

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4363 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.7.7`


Please check the `Fix Version/s` of ZENKO-4363, or the target
branch of this pull request.

Commit:
allow no parallelism for azure archive tests

Parallel runs are not supported in azure archive
tests as we can’t control what goes into the tar
files.

Issue: ZENKO-4363

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4363 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.7.8`


Please check the `Fix Version/s` of ZENKO-4363, or the target
branch of this pull request.

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4363 contains:


* `2.7.9`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.7.10`


Please check the `Fix Version/s` of ZENKO-4363, or the target
branch of this pull request.

Commit:
bump azurite

bert-e commented:
# Conflict




There is a conflict between your branch `improvement/ZENKO-4363-add-azure-archive-tests` and the
destination branch `development/2.7`.

Please resolve the conflict on **the feature branch** (`improvement/ZENKO-4363-add-azure-archive-tests`).

```sh
git fetch && \
git checkout origin/improvement/ZENKO-4363-add-azure-archive-tests && \
git merge origin/development/2.7
```

Resolve merge conflicts and commit

```sh
git push origin HEAD:improvement/ZENKO-4363-add-azure-archive-tests
```

bert-e commented:
# Conflict




There is a conflict between your branch `improvement/ZENKO-4363-add-azure-archive-tests` and the
destination branch `development/2.7`.

Please resolve the conflict on **the feature branch** (`improvement/ZENKO-4363-add-azure-archive-tests`).

```sh
git fetch && \
git checkout origin/improvement/ZENKO-4363-add-azure-archive-tests && \
git merge origin/development/2.7
```

Resolve merge conflicts and commit

```sh
git push origin HEAD:improvement/ZENKO-4363-add-azure-archive-tests
```

Commit:
bump CTST

Issue: ZENKO-4363-add-azure-archive-tests

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4363 contains:


* `2.7.11`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.7.12`


Please check the `Fix Version/s` of ZENKO-4363, or the target
branch of this pull request.

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4363 contains:


* `2.7.11`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.7.12`


Please check the `Fix Version/s` of ZENKO-4363, or the target
branch of this pull request.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

francoisferrand commented on tests/ctst/features/azureArchive.feature:
```diff
@@ -0,0 +1,129 @@
+    Scenario Outline: Respect maximum number of objects per archived Tar
+    Given a "<versioningConfiguration>" bucket
+    And a transition workflow to "e2e-azure-archive" location
+    And an object "obj1" of size <objectSize> bytes
```
could we add another parameter to this rule:

> And <count> objects "obj" of size <objectSize> bytes

where we could either run with multiple counts (→ see different behaviors), or at least avoid repeating 5 times the same step/condition ?

francoisferrand commented on tests/ctst/features/azureArchive.feature:
```diff
@@ -0,0 +1,129 @@
+        | versioningConfiguration | objectSize | packObjectCount |
+        |           Non versioned |          1 |               3 |
+        |               Versioned |          1 |               3 |
+        # |               Suspended |          1 |               3 |
```
should we have another config for "_non-versioned object in a versioned bucket_" ?

welansari replied:
i think we should wait for the follow up test ticket before adding this, as that case will change the semantics of the tests, we might want to make it it's own scenario

francoisferrand replied:
ok, please add this to the followup ticket so we don't forget

welansari replied:
done

francoisferrand commented on .github/scripts/end2end/configs/zenko.yaml:
```diff
@@ -94,6 +94,15 @@ spec:
       e2e-cold:
         debug: "true"
         command-timeout: "60s"
+    serverConfiguration:
+      ${AZURE_ARCHIVE_BACKEND_DESTINATION_LOCATION}:
```
with latest ZKOP :

```
server:
   tar:
      ...
   azure:
      ...
```

francoisferrand commented on solution/deps.yaml:
```diff
@@ -94,7 +94,7 @@ vault:
   sourceRegistry: registry.scality.com/zenko-operator
   image: zenko-operator
-  tag: 1.5.21
+  tag: 1.5.24
```
not needed, just rebase to get latest zkop...

francoisferrand commented:
/wait

welansari commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.7`



The following branches will **NOT** be impacted:

* `development/2.4`
* `development/2.5`
* `development/2.6`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.7`



The following branches have **NOT** changed:

* `development/2.4`
* `development/2.5`
* `development/2.6`



Please check the status of the associated issue ZENKO-4363.

Goodbye kerkesni.
